src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/LSPRequestInvoker.cs

```diff
@@ -19,5 +21,9 @@ internal abstract class LSPRequestInvoker
             LanguageServerKind serverKind,
             TIn parameters,
             CancellationToken cancellationToken);
+
+        internal abstract void AddClientNotifyAsyncHandler(AsyncEventHandler<LanguageClientNotifyEventArgs> clientNotifyAsyncHandler);
```

Comment:
Why have these here? Why not just have each class that depends on this functionality take the remote service broker as a constructor dependency?

Reply:
Refactored.